src/tdastro/effects/effect_model.py

```diff
@@ -20,19 +20,35 @@ class EffectModel:
     rest_frame : bool
         Whether the effect is applied in the rest frame of the observation (True)
         or in the observed frame (False).
+    effect_name : string
+        The name of the effect for logging and accessing parameters. If no name is
+        provided, the class name is used.
     parameters : dict
         A dictionary of parameters for the effect. Maps the parameter names to
         their setters.
     """
 
-    def __init__(self, rest_frame=True, **kwargs):
+    def __init__(self, *, rest_frame=True, effect_name=None, **kwargs):
         self.rest_frame = rest_frame
 
+        # Save the name of this effect.
+        if effect_name is None:
+            effect_name = self.__class__.__name__
+            effect_name = effect_name.replace(".", "_")
+        self.effect_name = effect_name
+
         # Automatically include all keyword arguments as settable parameters.
         self.parameters = {}
         for key, value in kwargs.items():
             self.add_effect_parameter(key, value)
 
+    def __str__(self):
+        return self.effect_name
+
+    def __repr__(self):
```

Comment:
```suggestion
    def __repr__(self):
        # Return parameter name only (and not value) to avoid unnecessarily long strings when chaining nodes.
```

It took me a minute to reason about this, so it might be nice to add a note to explain that we've already considered the pros and cons of a more typical repr output

Reply:
That is actually not the reason. Parameters can be dynamically sampled, so we do not know their value ahead of time.

Reply:
I added a comment about this.